src/lua/zencode_w3c.lua

```diff
@@ -20,88 +20,55 @@
 --on Wednesday, 14th July 2021
 --]]
 
-local function import_vm(obj, key)
-    if key == 'publicKeyBase64' then
-	return O.from_base64(obj)
-    elseif key == 'publicKeyBase58' then
-	return O.from_base58(obj)
-    else
-	return O.from_string(obj)
-    end
-end
-
-local function export_vm(obj, key)
-    if key == 'publicKeyBase64' then
-	return O.to_base64(obj)
-    elseif key == 'publicKeyBase58' then
-	return O.to_base58(obj)
-    else
-	return O.to_string(obj)
-    end
-end
-
 local function import_did_doc(doc)
     local res = {}
-    local vm = nil
     -- id must be always present in DID-documents
     ZEN.assert(doc.id, 'Invalid DID document: id not found')
     -- all the other fields are optional and imported as a string
-    -- except for verificationMethod that has its own import function
-    if doc.verificationMethod then
-	vm = ZEN.get(doc, 'verificationMethod', import_vm, tostring)
-	doc.verificationMethod = nil
-    end
     res = ZEN.get(doc, '.', O.from_string, tostring)
-    res.verificationMethod = vm
     return res
 end
 
 local function export_did_doc(doc)
     local res = {}
-    local vm = nil
-    if doc.verificationMethod then
-	vm = deepmap(export_vm, doc.verificationMethod)
-	doc.verificationMethod = nil
-    end
     res = deepmap(O.to_string, doc)
-    res.verificationMethod = vm
     return res
 end
 
 local function import_verification_method(doc)
```

Comment:
Some inspirational patterns: 
http://lua-users.org/wiki/SwitchStatement
and
https://stackoverflow.com/questions/37447704/what-is-the-alternative-for-switch-statement-in-lua-language